node/rpc/src/lib.rs

```diff
@@ -101,58 +94,67 @@ where
 		deny_unsafe,
 		is_authority,
 		network,
-		command_sink,
+		pending_transactions,
+		command_sink
 	} = deps;
 
-	io.extend_with(SystemApi::to_delegate(FullSystem::new(
-		client.clone(),
-		pool.clone(),
-		deny_unsafe,
-	)));
-	io.extend_with(TransactionPaymentApi::to_delegate(TransactionPayment::new(
-		client.clone(),
-	)));
-
-	// We currently don't want to support signing in the node. Users should prefer external tools
-	// for transaction signing. So just pass in an empty vector of signers.
-	let signers = Vec::new();
-	io.extend_with(EthApiServer::to_delegate(EthApi::new(
-		client.clone(),
-		pool.clone(),
-		moonbeam_runtime::TransactionConverter,
-		network.clone(),
-		signers,
-		is_authority,
-	)));
-	io.extend_with(NetApiServer::to_delegate(NetApi::new(
-		client.clone(),
-		network.clone(),
-	)));
-	io.extend_with(Web3ApiServer::to_delegate(Web3Api::new(client.clone())));
-	io.extend_with(EthPubSubApiServer::to_delegate(EthPubSubApi::new(
-		pool,
-		client,
-		network,
-		SubscriptionManager::<HexEncodedIdProvider>::with_id_provider(
-			HexEncodedIdProvider::default(),
-			Arc::new(subscription_task_executor),
-		),
-	)));
-
-	if let Some(command_sink) = command_sink {
-		io.extend_with(
-			// We provide the rpc handler with the sending end of the channel to allow the rpc
-			// send EngineCommands to the background block authorship task.
-			ManualSealApi::to_delegate(ManualSeal::new(command_sink)),
-		);
+	io.extend_with(
+		SystemApi::to_delegate(FullSystem::new(client.clone(), pool.clone(), deny_unsafe))
+	);
+	io.extend_with(
+		TransactionPaymentApi::to_delegate(TransactionPayment::new(client.clone()))
+	);
+
+	// TODO: are we supporting signing?
+	let mut signers = Vec::new();
+
+	io.extend_with(
+		EthApiServer::to_delegate(EthApi::new(
+			client.clone(),
+			pool.clone(),
+			moonbeam_runtime::TransactionConverter,
+			network.clone(),
+			pending_transactions.clone(),
+			signers,
+			is_authority,
+		))
+	);
+	io.extend_with(
+		NetApiServer::to_delegate(NetApi::new(
+			client.clone(),
+			network.clone(),
+		))
+	);
+	io.extend_with(
+		EthPubSubApiServer::to_delegate(EthPubSubApi::new(
+			pool.clone(),
```

Comment:
Clone is not necessary for `pool`, `client`, `network`

Reply:
Clippy should also catch this, right?

Reply:
Maybe not, but upstream uses `clone()`: https://github.com/paritytech/frontier/blob/fb76cd09c7b2fb59fd7bc0cd567de66839283305/template/node/src/rpc.rs#L129

Changing it would mean our code diverges a bit more from upstream. Thoughts?

Reply:
Yes, but that belongs to Frontier template example, which is not really upstream for us. In our case we should do whatever we consider right in our node.